docs/changelog/116026.yaml

```diff
@@ -0,0 +1,11 @@
+pr: 116026
+summary: Change Elasticsearch timeouts to 400 response instead of 5xx
+area: Infra/Core
+type: breaking
+issues: []
+breaking:
+  title: Change Elasticsearch timeouts to 400 response instead of 5xx
+  area: REST API
+  details: When a REST request times out, whether via a per-request timeout, or a system default, the request now returns a 400 response code.
```

Comment:
Can we enumerate all the API endpoints that are impacted?

I've tried several APIs with the `?timeout` param but was only able to produce two 5xx so far:
```
curl -v -XGET "http://localhost:9200/_cluster/health/my-index?wait_for_status=green&timeout=0s
--> 
status code: 408
res body: {"timed_out":true, ...}

curl -v -XPUT "http://localhost:9200/target-index/_mapping?timeout=0s" -H "Content-Type: application/json" -u elastic:changeme -d '{"properties": {"field3": {"type": "text"}}}'
status code: 200
res body: {"acknowledged": false}

curl -v -XPUT "http://localhost:9200/target-index-1?timeout=0" -H "Content-Type: application/json" -u elastic:changeme
status code: 200
res body: {"acknowledged": false, "shards_acknowledged": false}

curl -v -XPOST "http://localhost:9200/source-index/_clone/target-index?timeout=0" -H "Content-Type: application/json" -u elastic:changeme
status code: 200
res body: {"acknowledged": false, "shards_acknowledged": false}

GET /_tasks?actions=*&wait_for_completion=true&timeout=0s&pretty
status code: 200
res body: {
  "node_failures": [
    {
      "type": "failed_node_exception",
      "reason": "Failed node [01Z7rp-1QAKosLmFu2jOTQ]",
      "node_id": "01Z7rp-1QAKosLmFu2jOTQ",
      "caused_by": {
        "type": "receive_timeout_transport_exception",
        "reason": "[instance-0000000090][172.22.238.78:19761][cluster:monitor/tasks/lists[n]] request_id [396673064] timed out after [0ms]"
      }
    }, ... ]
}

GET /_tasks/QkCi4Xy3RTaSAaELiMSpTQ:3023?wait_for_completion=true&timeout=0ms&pretty
status code: 500
res body {
  "error" : {
    "root_cause" : [
      {
        "type" : "timeout_exception",
        "reason" : "timed out after [0s/0ms]"
      }
    ],
    "type" : "timeout_exception",
    "reason" : "timed out after [0s/0ms]"
  },
  "status" : 500
}

POST /test-index/_update_by_query?wait_for_completion=true&timeout=1ms&pretty
status code: 500
res body: {
  "took" : 85,
  "timed_out" : false, <-- misleading if you ask me
  "total" : 100,
  "updated" : 99,
  "deleted" : 0,
  "batches" : 1,
  "version_conflicts" : 0,
  "noops" : 0,
  "retries" : {
    "bulk" : 0,
    "search" : 0
  },
  "throttled_millis" : 0,
  "requests_per_second" : -1.0,
  "throttled_until_millis" : 0,
  "failures" : [
    {
      "index" : "test-index",
      "id" : "jIFEVJMBIY_Yr_cUNKcL",
      "cause" : {
        "type" : "mapper_exception",
        "reason" : "timed out while waiting for a dynamic mapping update"
      },
      "status" : 500
    }
  ]
}
```

Right now as a client of Elasticsearch it's really hard to gain confidence that this breaking change isn't impacting Kibana and I suspect our users will have similar anxiety when reading the release notes.